packages/clay-autocomplete/src/ClayAutocomplete.js

```diff
@@ -461,6 +481,16 @@ ClayAutocomplete.STATE = {
 	 * @type {?html}
 	 */
 	unstable_content: Config.any(),
+
+	/**
+	 * Validate the values of the input, use a Regex to validate the
+	 * input characters (e.g /[a-zA-Z0-9_]/g).
+	 * @default undefined
+	 * @instance
+	 * @memberof ClayAutocomplete
+	 * @type {?RegExp}
+	 */
+	validator: Config.instanceOf(RegExp),
```

Comment:
I think validator should be at least either a regexp or a function. Otherwise, we impose a very heavy restriction on the type of validation that can occur. For better flexibility, or `_runValidation` should also handle async values, so it should return a promise, which would be what's expected of the validator when a function is passed.

Reply:
Accepting a function can give you more flexibility... I do not know if it would be the best way to deal with async code here, these checks have to be done sync, since the Metal does not control the input, the user can typing and we have to make the changes so as not to lose information that the user has entered. Maybe this name is confusing, the idea is not to deal with it after the user has finished typing.

Reply:
Actually if we go with the idea of ​​validating the final value of the input, we must follow this path. I will include this functionality in another PR, but I will change the name of this API so it will not be confusing and improve the description.

Reply:
good compromise and way to move forward! 👍